src/Symfony/Component/Notifier/Bridge/OneSignal/Tests/OneSignalTransportTest.php

```diff
@@ -0,0 +1,162 @@
+use Symfony\Component\Notifier\Message\PushMessage;
+use Symfony\Component\Notifier\Message\SmsMessage;
+use Symfony\Component\Notifier\Test\TransportTestCase;
+use Symfony\Component\Notifier\Transport\TransportInterface;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+use Symfony\Contracts\HttpClient\ResponseInterface;
+
+/**
+ * @author Tomas Norkūnas <norkunas.tom@gmail.com>
+ */
+final class OneSignalTransportTest extends TransportTestCase
+{
+    /**
+     * @return OneSignalTransport
+     */
+    public function createTransport(?HttpClientInterface $client = null, ?string $recipientId = null): TransportInterface
+    {
+        return new OneSignalTransport('9fb175f0-0b32-4e99-ae97-bd228b9eb246', 'api_key', $recipientId, $client ?? $this->createMock(HttpClientInterface::class));
+    }
+
+    public function testCanSetCustomHost()
+    {
+        $transport = $this->createTransport();
+
+        $transport->setHost($customHost = self::CUSTOM_HOST);
+
+        $this->assertSame(sprintf('onesignal://9fb175f0-0b32-4e99-ae97-bd228b9eb246@%s', $customHost), (string) $transport);
+    }
+
+    public function testCanSetCustomHostAndPort()
+    {
+        $transport = $this->createTransport();
+
+        $transport->setHost($customHost = self::CUSTOM_HOST);
+        $transport->setPort($customPort = self::CUSTOM_PORT);
+
+        $this->assertSame(sprintf('onesignal://9fb175f0-0b32-4e99-ae97-bd228b9eb246@%s:%d', $customHost, $customPort), (string) $transport);
+    }
+
+    public function toStringProvider(): iterable
+    {
+        yield ['onesignal://9fb175f0-0b32-4e99-ae97-bd228b9eb246@onesignal.com', $this->createTransport()];
+        yield ['onesignal://9fb175f0-0b32-4e99-ae97-bd228b9eb246@onesignal.com?recipientId=ea345989-d273-4f21-a33b-0c006efc5edb', $this->createTransport(null, 'ea345989-d273-4f21-a33b-0c006efc5edb')];
+    }
+
+    public function supportedMessagesProvider(): iterable
+    {
+        yield [new PushMessage('Hello', 'World'), $this->createTransport(null, 'ea345989-d273-4f21-a33b-0c006efc5edb')];
+        yield [new PushMessage('Hello', 'World', (new OneSignalOptions())->recipient('ea345989-d273-4f21-a33b-0c006efc5edb'))];
+    }
+
+    public function unsupportedMessagesProvider(): iterable
+    {
+        yield [new SmsMessage('0611223344', 'Hello!')];
+        yield [new ChatMessage('Hello!')];
+        yield [$this->createMock(MessageInterface::class)];
+    }
+
+    public function testUnsupportedWithoutRecipientId()
+    {
+        $this->assertFalse($this->createTransport()->supports(new PushMessage('Hello', 'World')));
+    }
+
+    public function testSendThrowsWithoutRecipient()
+    {
+        $transport = $this->createTransport();
+
+        $this->expectException(LogicException::class);
+        $this->expectExceptionMessage('The "Symfony\Component\Notifier\Bridge\OneSignal\OneSignalTransport" transport should have configured `recipientId` via DSN or provided with message options.');
```

Comment:
This test is currently failing:

```
1) Symfony\Component\Notifier\Bridge\OneSignal\Tests\OneSignalTransportTest::testSendThrowsWithoutRecipient
Failed asserting that exception message 'The "Symfony\Component\Notifier\Bridge\OneSignal\OneSignalTransport" transport should have configured `defaultRecipientId` via DSN or provided with message options.' contains 'The "Symfony\Component\Notifier\Bridge\OneSignal\OneSignalTransport" transport should have configured `recipientId` via DSN or provided with message options.'.
```

I think the string just needs to be updated:

```suggestion
        $this->expectExceptionMessage('The "Symfony\Component\Notifier\Bridge\OneSignal\OneSignalTransport" transport should have configured `defaultRecipientId` via DSN or provided with message options.');
```